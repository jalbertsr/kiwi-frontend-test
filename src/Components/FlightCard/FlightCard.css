.spaceBetween {
  margin-bottom: 4%;
}

.priceTitle{
  text-align: center;
}