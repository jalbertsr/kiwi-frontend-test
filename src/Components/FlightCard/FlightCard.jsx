import React from 'react'
import { Link } from 'react-router-dom'
import { Card } from 'antd'

const FlightCard = ({ loading, title }) => (
  <Link to={`flights/${23}`}>
    <Card loading={loading} title={title}>
      Whatever content
    </Card>
  </Link>
)

export default FlightCard
