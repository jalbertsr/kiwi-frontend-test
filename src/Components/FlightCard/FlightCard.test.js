import FlightCard from './FlightCard'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import { MemoryRouter as Router } from 'react-router-dom'

configure({ adapter: new Adapter() })

/* eslint-disable no-unused-vars */

describe('FlightCard', () => {
  const props = {
    loading: true,
    title: 'From Barcelona to París',
    description: {
      flyFrom: 'BCN',
      flyTo: 'ORY',
      price: 285
    },
    departure: '22/10/2017',
    returning: '28/10/2017',
    index: 3
  }
  it('renders correctly', () => {
    const component = shallow(
      <Router>
        <FlightCard {...props} />
      </Router>)
    expect(component).toMatchSnapshot()
  })
  it('renders a "Price: "', () => {
    const component = shallow(
      <Router>
        <FlightCard {...props} />
      </Router>)
    expect(component.html().includes(
      '<div class="ant-card-head-title">From Barcelona to París</div>')
    ).toBe(true)
  })
})
