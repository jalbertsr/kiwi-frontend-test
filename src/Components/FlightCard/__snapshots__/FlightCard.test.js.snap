// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightCard renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter>
    <FlightCard
      departure="22/10/2017"
      description={
        Object {
          "flyFrom": "BCN",
          "flyTo": "ORY",
          "price": 285,
        }
      }
      index={3}
      loading={true}
      returning="28/10/2017"
      title="From Barcelona to París"
    />
  </MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <FlightCard
        departure="22/10/2017"
        description={
          Object {
            "flyFrom": "BCN",
            "flyTo": "ORY",
            "price": 285,
          }
        }
        index={3}
        loading={true}
        returning="28/10/2017"
        title="From Barcelona to París"
      />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "a1gi83",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "a1gi83",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "departure": "22/10/2017",
        "description": Object {
          "flyFrom": "BCN",
          "flyTo": "ORY",
          "price": 285,
        },
        "index": 3,
        "loading": true,
        "returning": "28/10/2017",
        "title": "From Barcelona to París",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <FlightCard
          departure="22/10/2017"
          description={
            Object {
              "flyFrom": "BCN",
              "flyTo": "ORY",
              "price": 285,
            }
          }
          index={3}
          loading={true}
          returning="28/10/2017"
          title="From Barcelona to París"
        />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "a1gi83",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "a1gi83",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "departure": "22/10/2017",
          "description": Object {
            "flyFrom": "BCN",
            "flyTo": "ORY",
            "price": 285,
          },
          "index": 3,
          "loading": true,
          "returning": "28/10/2017",
          "title": "From Barcelona to París",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
